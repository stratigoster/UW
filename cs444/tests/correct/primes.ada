-- generates the first i prime numbers
package main is
  procedure genprimes(i : integer);
  function isprime(i : integer) return boolean;
end main;

package body main is
  -- check if a number is prime
  function isprime(i : integer) return boolean is
  begin
    if i < 0 then
      i := abs(i);
    end if;
    if i = 0 or else i = 1 then
      return false;
    elsif i = 2 then
      return true;
    else
      for zzz in 2..i-1 loop
        if i mod zzz = 0 then
          return false;
        end if;
      end loop;
      return true;
    end if;
  end isprime;
  -- generate the first i prime numbers
  procedure genprimes(i : integer) is
  begin
    declare
      zzz : integer := 1;
    begin
      while i > 0 loop
        if isprime(zzz) then
          i := i-1;
          write(zzz);
          write(" ");
        end if;
        zzz := zzz + 1;
      end loop;
    end;
  end genprimes;

  num : integer := 0;

begin
  write("Please enter a number: ");
  read(num);
  write("\nThe first ");
  write(num);
  write(" prime numbers are:\n");
  genprimes(num);
  write("\n");
end main;
