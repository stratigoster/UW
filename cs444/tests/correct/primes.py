#!/usr/bin/env python
import math
def _forwardGen(seq):
	for i in seq:
		yield { 'all':i }
def _reverseGen(seq):
	i = len(seq)
	while i>0:
		i = i - 1
		yield { 'all':seq[i] }
class _dummy_(Exception):pass
class _5(Exception):pass
class _3(Exception):pass
class _numeric_error(Exception):pass
class _constraint_error(Exception):pass
class main:
	try:
		def _2_isprime(self,i):
			try:
				pass
				if (i['all'] < 0):
					i['all'] = abs(i['all']) 
				if ((i['all'] == 0) or (i['all'] == 1)):
					return 0
				elif (i['all'] == 2):
					return 1
				else:
					try:
						for zzz in _forwardGen(range(2,(i['all'] - 1)+1)):
							if ((i['all'] % zzz['all']) == 0):
								return 0
					except _dummy_:
						pass
					return 1
			except _dummy_:
				pass
		def _1_genprimes(self,i):
			try:
				pass
				def _4():
					try:
						pass
						zzz = { 'all':1 }
						try:
							while (i['all'] >0):
								pass
								if self._2_isprime(i = {'all':zzz['all']}):
									i['all'] = (i['all'] - 1) 
									print zzz['all'],
									print ' ',
								zzz['all'] = (zzz['all'] + 1) 
						except _5:
							pass
					except _dummy_:
						pass
				_4()
			except _dummy_:
				pass
		num = { 'all':0 }
		def __init__(self):
			try:
				pass
				print 'Please enter a number: ',
				self.num['all'] = int(input()) 
				print '\nThe first ',
				print self.num['all'],
				print ' prime numbers are:\n',
				self._1_genprimes(i = {'all':self.num['all']})
				print '\n',
			except _dummy_:
				pass
	except _dummy_:
		pass
main()
