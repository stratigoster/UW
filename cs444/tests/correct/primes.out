-== 273 ==- npow@cpu16 -== ~/compilers/docs/correct ==-
--> ./primes.exe
Please enter a number:  10

The first  10  prime numbers are:
2   3   5   7   11   13   17   19   23   29

